### ISSUE TITLE
Version 2.5.0: Entrant can partially enter without selecting style dropdown 

### ISSUE THREAD

**Issue by jimmyorkin** [geoffhumphrey/brewcompetitiononlineentry#1397]
**Verify the Following**
Reporting About the Latest Version? Yes

Searched Issue Reports for Bug? Yes

Reviewed the Good To Know Issue Reports? Yes

**BCOE&M Info**
Version: 2.5.0 latest commit 
Installation URL: bcoem.bluebonnetbrewoff.org
Hosted Installation? No

**Describe the Bug**
The add entry page: "https://localhost/bcoem250/index.php?section=brew&action=add" allows the entrant to enter the entry name and click the add entry button. A row is entered in the brewing table but the brewStyle, brewCategory columns are NULL. The brewCategorySort has '00' and brewSubCategory has an 'A'.

The entrant gets this message after clicking Add Entry at the top of the page. This entry cannot be accepted or confirmed until a style has been chosen. Unconfirmed entries may be deleted from the system without warning.

A bit panicked here. I believe an entrant could pay for one of these partial in entries.

**To Reproduce**
1. Go to enter and entry page.
2. Enter a Name
3. Click Add Entry.

**Expected Behavior**
The Add Entry button to not be enabled until a style is selected.

**Comment by geoffhumphrey** [comment]
Hey Jimmy, just in case this does happen on a live site, there's fallback scripting in place to prevent anyone from paying for any unconfirmed entries. All "pay" buttons and/or links will be disabled or hidden on the brewer's My Account page. 

If the entrant does navigate to the pay entries screen, scripting there will not factor in any "unconfirmed" entries and a notice at the top will alert the entrant, like so:

![image](https://user-images.githubusercontent.com/5216502/220175693-418503d6-cd72-420b-9f05-93c894efaff9.png)

This way, for comps like Blue Bonnet who require payment immediately upon entering, entrants won't be able to "steal" a entry slot away from someone else.

Of course, your solution is applicable as well.

Hope that helps!

**Comment by jimmyorkin** [comment]
Geoff,

That is good. I expected the JavaScript in the page to know that the style had not been selected and not enable the save button.

I don't know much about HTML or JavaScript so pardon my ignorance. I see this the HTML served to my browser for the add entry page "onSubmit="return CheckRequiredFields()">" I assume there is supposed to be a function CheckRequiredFields() but I do not see it in the page. Should it be checking that the style has been selected before enabling the button?

**Comment by geoffhumphrey** [comment]
Hi @jimmyorkin - good catch... the CheckRequiredFields function isn't used and/or declared anymore. It was deprecated in favor of using another, more robust form validation script. I've removed any calls to that function entirely.

I noodled around with things and I have updated scripting that prevents users from submitting an entry without data in any required field, including Style, making it obvious which fields are required (indicated with in red if users select the Add Entry button before they are filled out).
